### ISSUE TITLE
oauth2 plugin breaking json request

### ISSUE THREAD

**Issue by arsdehnel** [Kong/kong#2799]
### Summary

Adding a new API to our Kong setup that already has a few APIs on it although this is our first that is a nodejs application.  Might be that we're just not using the right set of middleware to parse things.  When we make the request without the OAuth 2 plugin enabled (Client Credentials) for the endpoint it's working fine.  But when we enable that plugin and make the exact same request we're getting:

```
8/12/2017 6:36:14 AMSyntaxError: Unexpected token %
8/12/2017 6:36:14 AM    at parse (/usr/src/app/node_modules/body-parser/lib/types/json.js:83:15)
8/12/2017 6:36:14 AM    at /usr/src/app/node_modules/body-parser/lib/read.js:116:18
8/12/2017 6:36:14 AM    at invokeCallback (/usr/src/app/node_modules/raw-body/index.js:262:16)
8/12/2017 6:36:14 AM    at done (/usr/src/app/node_modules/raw-body/index.js:251:7)
8/12/2017 6:36:14 AM    at IncomingMessage.onEnd (/usr/src/app/node_modules/raw-body/index.js:307:7)
8/12/2017 6:36:14 AM    at emitNone (events.js:105:13)
8/12/2017 6:36:14 AM    at IncomingMessage.emit (events.js:207:7)
8/12/2017 6:36:14 AM    at endReadableNT (_stream_readable.js:1045:12)
8/12/2017 6:36:14 AM    at _combinedTickCallback (internal/process/next_tick.js:138:11)
8/12/2017 6:36:14 AM    at process._tickCallback (internal/process/next_tick.js:180:9)
```

### Steps To Reproduce

1. Disable OAuth 2 
2. Request to `POST myapi.com/path/to/resource`:

`curl -H "Content-Type: application/json" -d '{"test": "this"}' https://myapi.com/path/to/resource`

3. See parsed JSON in the logs
4. Enable OAuth 2
5. Make request to `GET myapi.com/path/to/resource/oauth2/token` to get the access token
6. Take that token and make a request to `POST myapi.com/path/to/resource?access_token=<token>`
7. Check logs and see the error.

### Additional Details & Logs

- Kong version: 0.9.9
- Operating System: Docker image

**Comment by arsdehnel** [comment]
Been staring at this for 2 hours and of course as soon as I post an issue I got it working.  Not sure why I'd get the error I'm getting though, that seems weird to me.  First off what is working:

1. Removed the `access_token` from the query string since I'm doing a POST
2. Added `-X POST` because that's pretty important
3. Put the access token into the Authorization header

```
curl -H "Content-Type: application/json" -H "Authorization: Bearer <access_token>" -d '{"test": "this"}' -X POST myapi.com/path/to/resource
```

But seems like if I do a `POST` to an endpoint that has client credentials OAuth enabled but miss the Authorization header that it should return some sort of unauthorized response.  Instead it appears to be sort of passing _something_ through even though the token isn't in the right place but it's not the original request body because parsing it as json gets me the above unexpected token error.
